src/mpid/ch4/generic/am/mpidig_am_init.c

```diff
@@ -70,6 +127,17 @@ int MPIDIG_am_init(void)
                                               MPIDIU_REQUEST_POOL_MAX_NUM_CELLS, host_alloc,
                                               host_free, &MPIDI_global.request_pool);
     MPIR_ERR_CHECK(mpi_errno);
+    /* The cell size need to match the send side (ofi short msg size) */
+    mpi_errno = MPIDU_genq_private_pool_create_unsafe(MPIR_CVAR_CH4_AM_PACK_BUFFER_SIZE,
+                                                      MPIR_CVAR_CH4_NUM_AM_PACK_BUFFERS_PER_CHUNK,
+                                                      INT_MAX,
+                                                      host_alloc_buffer_registered,
+                                                      host_free_buffer_registered,
+                                                      &MPIDI_global.unexp_pack_buf_pool);
+    MPIR_ERR_CHECK(mpi_errno);
+    MPIDU_genq_private_pool_set_warning_limit(MPIDI_global.unexp_pack_buf_pool,
+                                              MPIR_CVAR_CH4_MAX_AM_UNEXPECTED_PACK_BUFFERS_SIZE_BYTE);
```

Comment:
I don't particularly like this function.  It's very specific to this use case.  I'd recommend simply dropping this functionality.

Reply:
I see, then for the size warning, I can just tracking it inside CH4.

Reply:
I don't think there's an easy way to track it.  That's why I recommended that you simply drop that functionality.

Reply:
dropped